### PR TITLE
feat(netlify): add support for image.remotePatterns and images.domains with Netlify Image CDN

### PR DESCRIPTION
**Pull request by ascorbic**
## Changes

Previously the Netlify adapter was not able to enable support for remote images. This PR adds support for `image.remotePatterns` and `images.domains` config options. If a user sets these then the adapter converts them into regexes and writes them to Netlify's `images.remote_images` config option.

## Testing

Added a test image to the fixture site. Added tests for generating the patterns and the config file.

## Docs

Docs PR: https://github.com/withastro/docs/pull/7375

### CODE REVIEW

**Comment by alexanderniebuhr** [comment]
@ascorbic thanks for the PR, we'll get it reviewed asap.. Once you have the docs PR, please crosslink both PRs :)

**Comment by ascorbic** [comment]
Thanks! Docs are now ready

**Review by Princesseuh — APPROVED** [review]
Can't speak too much to the Netlify part of this, but the Astro part looks fine. Thank you for adding it, I like our `domains` and `remotePatterns` options working automatically for every services, it's quite cool.

File: .changeset/nine-lamps-cry.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/netlify': minor
+---
+
+Adds support for image.remotePatterns and images.domains with Netlify Image CDN
```
Comment (alexanderniebuhr) [inline .changeset/nine-lamps-cry.md:5]:
```suggestion
Adds support for `image.remotePatterns` and `images.domains` with Netlify Image CDN
```

Not sure but maybe it looks better as inline code block

**Review by alexanderniebuhr — APPROVED** [review]
LGTM

File: packages/netlify/src/index.ts [hunk 2]
```diff
@@ -24,6 +24,90 @@ const isStaticRedirect = (route: RouteData) =>
...
+}
+
+async function writeNetlifyDeployConfig(config: AstroConfig) {
+	const remoteImages: Array<string> = [];
+	// Domains get a simple regex match
+	remoteImages.push(
+		...config.image.domains.map((domain) => `https?:/\/${domain.replaceAll('.', '\\.')}\/.*`)
+	);
+	// Remote patterns need to be converted to regex
+	remoteImages.push(...config.image.remotePatterns.map(remotePatternToRegex));
```
Comment (ematipico) [inline packages/netlify/src/index.ts:98]: Isn't there a way to check if the regex isn't malformed before using it? After all it's built dynamically with a lot of branching

Comment (ascorbic) [reply packages/netlify/src/index.ts:98]: That's a good idea

Comment (ascorbic) [reply packages/netlify/src/index.ts:98]: I guess the question is what we do with that information if it is malformed. Throw? Log a warning?

Comment (ematipico) [reply packages/netlify/src/index.ts:98]: I suppose we can log an error/warning to the user, and maybe logging the produced regex will help to identify a possible bug. I don't think it's worth blocking the build, right?

Comment (ascorbic) [reply packages/netlify/src/index.ts]: @ematipico I've added a warning log. See if it looks right to you

**Comment by ascorbic** [comment]
Is there anything else that needs doing before this can go in?

**Review by ematipico — APPROVED** [review]

**Comment by ascorbic** [comment]
Thanks, @ematipico. What's the timeline for these releases to go out?

**Comment by alexanderniebuhr** [comment]
@ascorbic I just released it!

**Comment by ascorbic** [comment]
That's great! Thanks

**Pull request merged**